packages/odyssey-react-mui/src/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,757 @@
+   */
+  hasPagination?: boolean;
+  /**
+   * If true, the table will include checkboxes on each row, enabling
+   * the user to select some or all rows.
+   */
+  hasRowSelection?: boolean;
+  /**
+   * If true, the global table search controls will be shown.
+   */
+  hasSearch?: boolean;
+  /**
+   * If true, the end user can sort columns (ascending, descending, or neither)
+   */
+  hasSorting?: boolean;
+  /**
+   * If true, the end user can reorder rows via a drag-and-drop interface
+   */
+  hasRowReordering?: boolean;
+  /**
+   * If true, the search field will include a Search button, rather than
+   * firing on input change.
+   */
+  hasSearchSubmitButton?: boolean;
+  /**
+   * The debounce time, in milliseconds, for the search input firing
+   * `onChangeSearch` when changed. If `hasSearchSubmitButton` is true,
+   * this doesn't do anything.
+   */
+  searchDelayTime?: number;
+  /**
+   * Callback that fires when a row (or rows) is selected or unselected.
+   */
+  onChangeRowSelection?: (rowSelection: DataTableRowSelectionState) => void;
+  /**
+   * Callback that fires whenever the table needs to fetch new data, due to changes in
+   * page, results per page, search input, filters, or sorting
+   */
+  getData: ({
+    page,
+    resultsPerPage,
+    search,
+    filters,
+    sort,
+  }: DataTableGetDataType) =>
+    | MRT_TableOptions<MRT_RowData>["data"]
+    | Promise<MRT_TableOptions<MRT_RowData>["data"]>;
+  /**
+   * Callback that fires when the user reorders rows within the table. Can be used
+   * to propogate order change to the backend.
+   */
+  onReorderRows?: ({ rowId, newRowIndex }: DataTableOnReorderRowsType) => void;
+  /**
+   * The current page number.
+   */
+  currentPage?: number;
+  /**
+   * The number of results per page.
+   */
+  resultsPerPage?: number;
+  /**
+   * The type of pagination controls shown. Defaults to next/prev buttons, but can be
+   * set to a simple "Load more" button by setting to "loadMore".
+   */
+  paginationType?: (typeof paginationTypeValues)[number];
+  /**
+   * Action buttons to display in each row
+   */
+  rowActionButtons?: DataTableRowActionsProps["rowActionButtons"];
+  /**
+   * Menu items to include in the optional actions menu on each row.
+   */
+  rowActionMenuItems?: DataTableRowActionsProps["rowActionMenuItems"];
+  /**
+   * Menu items to include in the bulk actions menu, which appears above the table if a row or rows are selected
+   */
+  bulkActionMenuItems?: (
+    selectedRows: MRT_RowSelectionState,
+  ) => MenuButtonProps["children"];
+  /**
+   * If `error` is not undefined, the DataTable will indicate an error.
+   */
+  errorMessage?: string;
+  /**
+   * The component to display when the table is displaying the initial empty state
+   */
+  emptyPlaceholder?: ReactElement<typeof DataTableEmptyState>;
+  /**
+   * The component to display when the query returns no results
+   */
+  noResultsPlaceholder?: ReactElement<typeof DataTableEmptyState>;
```

Comment:
I believe this should be a ReactNode. Why are we attempting to be restrictive here? ReactElement<typeof DataTableEmptyState> won't actually limit someone to only providing something of type DataTableEmptyState from what I know?